### ISSUE TITLE
[Balance Suggestion] - Mimic/normal chest shouldn't be determined every time a generated chest is opened

### ISSUE THREAD

**Issue by SkyArrow76** [Sollace/Unicopia#343]
The way that mimics appear seems like it needs a little tweaking.
If I've already opened a chest, and taken an item, it shouldn't turn into a mimic when I open it another time.
The only way to de-mimic is to open until it turns into a mimic, kill it, and place the chest.
I feel like mimics should be determined upon generation, not every time a generated chest is opened.
Cause right now they're just a re-rollable Schrodinger's Cat

**Comment by Sollace** [comment]
The mimic is only determined when you open a chest that hasn't generated loot. (if it has a loot table assigned). Obviously if you have mods that cause chests to re-roll loot, it will cause this problem.

**Comment by SkyArrow76** [comment]
Weird.....I don't believe I have any mods that would do that. To test this happening, I used a very light list of mods:![image](https://github.com/Sollace/Unicopia/assets/33587750/52d8d685-8d37-4ea5-be97-ae19736fac54)

I've got a recording of it happening too. 

https://github.com/Sollace/Unicopia/assets/33587750/92e5442a-8c68-4508-8c0b-ed9cedc0ae96


For the first chest, Its normal. even after I take an item out. If I keep opening it, though, it turns into a mimic.
For the other chest, I just kept opening it and closing it, and got it to be a mimic.
Obviously, you wouldn't normally spam open chests like that, but I've had it happen where I took some items out and had a full inventory, so dumped some, and when I opened the same chest again to get more out, it was a mimic.

Maybe that's intended gameplay though?
